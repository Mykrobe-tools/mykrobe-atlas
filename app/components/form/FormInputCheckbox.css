.wrap {
  composes: wrap from 'styles/form.css';
}

.label {
  composes: label from 'styles/form.css';
}

.items {
  display: flex;
  flex-wrap: wrap;
  vertical-align: top;
}

.item {
  padding-right: 24px;
  color: var(--color-grey-mid);
}

.item:last-child {
  padding-right: 0;
}

.input {
  margin-right: 8px;
  cursor: pointer;
}
