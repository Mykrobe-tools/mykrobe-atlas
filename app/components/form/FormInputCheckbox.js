/* @flow */

import React, { Component } from 'react';

import FormLabel from './FormLabel';

class FormInputCheckbox extends Component {

  constructor(props: Object) {
    super(props);
  }

  render() {
    return (
      <div>
        <FormLabel
          htmlFor={this.props.name}
          label={this.props.title} />
        <div>
          {this.props.options.map(opt => {
            return (
              <FormLabel
                key={opt.value}
                label={opt.label}>
                <input
                  name={this.props.name}
                  onChange={this.props.onChange.bind(this)}
                  value={opt.value}
                  checked={this.props.selectedOptions.indexOf(opt.value) > -1}
                  type="checkbox" />
              </FormLabel>
            );
          })}
        </div>
      </div>
    );
  }
}

FormInputCheckbox.propTypes = {
  title: React.PropTypes.string.isRequired,
  name: React.PropTypes.string.isRequired,
  options: React.PropTypes.array.isRequired,
  selectedOptions: React.PropTypes.array,
  onChange: React.PropTypes.func.isRequired
}

export default FormInputCheckbox;
