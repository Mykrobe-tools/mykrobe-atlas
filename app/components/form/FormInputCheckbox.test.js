import * as React from 'react';
import renderer from 'react-test-renderer';
import FormInputCheckbox from './FormInputCheckbox';

describe('FormInputCheckbox component snapshot', () => {
  const name = 'loremIpsum';
  const title = 'Lorem Ipsum';
  const onChange = jest.fn();

  it('renders correctly with a single input', () => {
    const selectedOptions = [];
    const options = [{ value: 'lorem', label: 'Lorem' }];
    const component = renderer.create(
      <FormInputCheckbox
        name={name}
        title={title}
        options={options}
        selectedOptions={selectedOptions}
        onChange={onChange}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple inputs', () => {
    const selectedOptions = ['lorem', 'dolorSitAmet'];
    const options = [
      { value: 'lorem', label: 'Lorem' },
      { value: 'ipsum', label: 'Ipsum' },
      { value: 'dolorSitAmet', label: 'Dolor Sit Amet' },
    ];
    const component = renderer.create(
      <FormInputCheckbox
        name={name}
        title={title}
        options={options}
        selectedOptions={selectedOptions}
        onChange={onChange}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
