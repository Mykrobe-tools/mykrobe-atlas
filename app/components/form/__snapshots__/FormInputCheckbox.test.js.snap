// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputCheckbox component snapshot renders correctly with a single input 1`] = `
<div
  className="wrap"
>
  <div
    className="label"
  >
    <label
      className="label"
      htmlFor="loremIpsum"
    >
      Lorem Ipsum
    </label>
  </div>
  <div
    className="items"
  >
    <div
      className="item"
    >
      <label
        className="label"
      >
        <input
          checked={false}
          className="input"
          name="loremIpsum"
          onChange={[Function]}
          type="checkbox"
          value="lorem"
        />
        Lorem
      </label>
    </div>
  </div>
</div>
`;

exports[`FormInputCheckbox component snapshot renders correctly with multiple inputs 1`] = `
<div
  className="wrap"
>
  <div
    className="label"
  >
    <label
      className="label"
      htmlFor="loremIpsum"
    >
      Lorem Ipsum
    </label>
  </div>
  <div
    className="items"
  >
    <div
      className="item"
    >
      <label
        className="label"
      >
        <input
          checked={true}
          className="input"
          name="loremIpsum"
          onChange={[Function]}
          type="checkbox"
          value="lorem"
        />
        Lorem
      </label>
    </div>
    <div
      className="item"
    >
      <label
        className="label"
      >
        <input
          checked={false}
          className="input"
          name="loremIpsum"
          onChange={[Function]}
          type="checkbox"
          value="ipsum"
        />
        Ipsum
      </label>
    </div>
    <div
      className="item"
    >
      <label
        className="label"
      >
        <input
          checked={true}
          className="input"
          name="loremIpsum"
          onChange={[Function]}
          type="checkbox"
          value="dolorSitAmet"
        />
        Dolor Sit Amet
      </label>
    </div>
  </div>
</div>
`;
